scheduler/pkg/agent/server_test.go

```diff
@@ -1062,9 +1066,9 @@ func TestSubscribe(t *testing.T) {
 				}(idx, s)
 			}
 
-			time.Sleep(10 * time.Second)
-
-			g.Expect(len(server.agents)).To(Equal(test.expectedAgentsCountAfterClose))
+			for len(server.agents) != test.expectedAgentsCountAfterClose {
```

Comment:
this will just sleep forever if the test fails, right?

Reply:
yes thats true, I can also cap it to a max count?

Reply:
done in https://github.com/SeldonIO/seldon-core/pull/6022/commits/dfb347230c479062d310d00f1edce4e1df41b6e4